ci/integration/conf_scheduler.yaml

```diff
@@ -0,0 +1,23 @@
+---
+
+pluginDir: /usr/lib64/sg-core
+logLevel: debug
+handlerErrors: false
+
+transports: []
+applications:
+  - name: scheduler
+    config:
+      schedule:
+        - name: test1
+          command: "echo 'test1'"
```

Comment:
I'm not sure in this exact situation, but I know sometimes it might be a security issue to execute shell commands read from a config file like this, which it looks like you are trying to do here (I think we talked about this on one of our meetings). If an attacker can somehow edit the file he could execute any script he wants and gain access to the pod, but maybe I'm just too paranoid :D. @csibbitt I think you have some background in security?

Reply:
That is true indeed. That's how availability monitoring frameworks work unfortunately :). Security will be handled in production by isolation of sg-agent in it's own container and restriction of execution permission via sudo config the same way as it is now handled with collectd-sensubility (executor plugin will run on overcloud nodes). In other words, by default sg-agent will be able to execute just the non-root commands + whatever you will allow it by sudo config.

Reply:
Didn't we talk in the happy hour that we would rather have predefined jobs set in the executor config rather than sending arbitrary scripts over the network? I'm picturing the situation where this scheduler is running on server side sending commands to the agent on the client side over the network. Even in a container without sudo, an attacker could execute malicious code against the OSP APIs and ruin a network, or post internal network info back out

Reply:
In practice I do not know how often someone would be running that configuration, but we did discuss that being an option

Reply:
> Didn't we talk in the happy hour that we would rather have predefined jobs set in the executor config rather than sending arbitrary scripts over the network? 

We did yes, that's one of a possible use case. Right now this is not possible yet, since app plugins don't have access to transport plugins in sense that app could ask transport to send message over the bus, but it is in TODO list.

> Even in a container without sudo, an attacker could execute malicious code against the OSP APIs,

The attacker would first have to have access to the correct required API nodes. Allowing openstack command to root only and handling the command access via sudo is still possible.

> or post internal network info back out

If the attacker has access to configuration on STF server side to change check command, he probably also then have access to the whole data set being stored in there. That basically means he already have that kind of data IMO.

We can of course assume that person(s) maintaining cloud solution and it's monitoring does not have at least average IQ, but that would slip us to over-engineering simple things :) .

Reply:
We don't do authentication against the AMQP bus though, so if an attacker gets access to it (say from another compromised pod), they could still spoof messages. You're right there are much bigger problems if an attacker gets root access to a container, this would just be about minimizing the damage. Chris' security updates will also solve this problem I believe. Okay this works for me